opencensus/stats/exporters/stackdriver_exporter.py

```diff
@@ -0,0 +1,337 @@
+		every metric created by this
+		exporter in Stackdriver Monitoring.
+
+		If unset, this defaults to a single label with key "opencensus_task" and
+		value "py-<pid>@<hostname>". This default ensures that the set of labels
+		together with the default Resource (global) are unique to this
+		process, as required by Stackdriver Monitoring.
+
+		If you set default_monitoring_labels, make sure that the Resource field
+		together with these labels is unique to the
+		current process. This is to ensure that there is only a single writer to
+		each TimeSeries in Stackdriver.
+
+		Set this to Labels to avoid getting the
+		default "opencensus_task" label. You should only do this if you know that
+		the Resource you set uniquely identifies this Python process. 
+		"""
+		return self._default_monitoring_labels
+
+class StackdriverStatsExporter(base.StatsExporter):
+	""" StackdriverStatsExporter exports stats to the Stackdriver Monitoring. """
+	def __init__(self, 
+				options = Options(), 
+				client = None,   
+				default_labels = {}): # Add here an argument for transport object
+		self._options = options
+		self._client = client # Add here a property for transport object
+		self._default_labels = default_labels
+
+	@property
+	def options(self):
+		return self._options
+
+	@property
+	def client(self):
+		return self._client
+
+	@property
+	def default_labels(self):
+		return self._default_labels
+
+	def set_default_labels(self, value):
+			self._default_labels = value
+
+	def on_register_view(self, view):
+		""" create metric descriptor for the registered view  """
+		if view:
+			self.create_metric_descriptor(view)
+
+	def emit(self, view_datas):
+		""" export data to Stackdriver Monitoring """
+		if view_datas:
+			self.handle_upload(view_datas)
+
+	def export(self, view_datas):
+		""" export data to transport class """
+		if view_datas:
+			pass #self.transport.export(view_datas) || Uncomment this when transport class gets merged to this branch
+
+	def handle_upload(self, view_datas):
+    	""" handle_upload handles uploading a slice of Data, 
+		as well as error handling. 
+		"""
+		if view_datas:
+			self.upload_stats(view_datas)
+		
+	def upload_stats(self, view_datas):
+		""" It receives an array of view_data object 
+			and create time series for each value
+		"""
+		for request in self.make_request(view_datas, 
+							MAX_TIME_SERIES_PER_UPLOAD):
+			self.client.create_time_series(request[CONS_NAME],
+											request[CONS_TIME_SERIES])
+
+	def make_request(self, view_datas, limit):
+		""" Create the data structure that will be 
+			sent to Stackdriver Monitoring
+		"""
+		requests = []
+		time_series = []
+
+		resource = self.options.resource
+		for view_data in view_datas:
+			series = monitoring_v3.types.TimeSeries()
+			series.metric.type = namespaced_view_name(view_data.view.name)
+
+			if not resource:
+    				series.resource.type = 'global'
+			else:
+				series.resource = resource
```

Comment:
Move this part to a separate function called `def create_time_series(...args needed here)` would be better.